src/nest.cpp

```diff
@@ -0,0 +1,24 @@
+#include <Rcpp.h>
+#include <dplyr.h>
+using namespace Rcpp;
+
+// [[Rcpp::export]]
+SEXP extract_groups(const List data, const CharacterVector nested_cols) {
+  ListOf<IntegerVector> indices(data.attr("indices"));
```

Comment:
Is this faster than the R equivalent using the indices attribute? The reason I didn't take this approach original is that sometimes indices are lazily computed - i.e. after you `filter()` a grouped data frame, the indices are dropped, and only recomputed when needed (IIRC)